python-avd/pyavd/_eos_designs/structured_config/base/utils.py

```diff
@@ -51,25 +66,13 @@ def _build_source_interfaces(
             # Check for duplicate VRF
             # inband_mgmt_vrf returns None in case of VRF "default", but here we want the "default" VRF name to have proper duplicate detection.
             inband_mgmt_vrf = self.shared_utils.inband_mgmt_vrf or "default"
-            if [source_interface for source_interface in source_interfaces if source_interface.get("vrf", "default") == inband_mgmt_vrf]:
+            if [source_interface for source_interface in source_interfaces if source_interface.vrf or inband_mgmt_vrf == "default"]:
```

Comment:
I don't think this logic change is correct (though it is not failing any test so need to dig deeper):
* beforehand we were checking is source_interface.vrf (defaulting to default) was equal to inband_mgmt_vrf 
* now we check if there is a source_interface.vrf (any value) OR if the inband_mgmt_vrf is default

Reply:
I think there is missing a paranthesis pair around the or.

Reply:
fixed